### PR TITLE
Dala 4165 datamodel changes

### PR DESCRIPTION
**Pull request by aliabdoli77**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [ ] Documentation is updated as required
- [ ] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [x] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [x] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [x] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [x] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [ ] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [ ] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/documents/CompanyReport.kt [hunk 1]
```diff
@@ -19,11 +18,5 @@ data class CompanyReport(
...
     override val fileName: String? = null,
 
-    // The following annotation is required due to a known issue with the openApi generator for fields starting with is
-    @field:JsonProperty()
-    val isGroupLevel: YesNoNa? = null,
-
-    val reportDate: LocalDate? = null,
-
-    val currency: String? = null,
+    val publicationDate: LocalDate? = null,
```
Comment (jhaerDfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/documents/CompanyReport.kt:21]: warum wird der company report angepasst? (keine info im jira ticket)

File: dataland-frontend/tests/component/components/resources/dataRequest/ReviewRequestButtons.cy.ts [hunk 2]
```diff
@@ -20,6 +20,9 @@ describe("Component tests for the data request review buttons", function (): voi
   before(() => {
     cy.fixture("DataRequestsMock").then((jsonContent) => {
       mockedRequests = jsonContent as Array<StoredDataRequest>;
+      if (mockedRequests[0].reportingPeriod === "2022") {
+        mockedRequests[0].reportingPeriod = "2024";
+      }
```
Comment (jhaerDfine) [inline dataland-frontend/tests/component/components/resources/dataRequest/ReviewRequestButtons.cy.ts:25]: warum denn dieses if?

File: dataland-frontend/src/components/resources/dataTable/MultiLayerDataTable.vue [hunk 3]
```diff
@@ -55,86 +55,64 @@
...
-    },
-  },
-});
+    }
+  }
+  const mostRecentSourceToolTip = latestDate
+    ? `Publication date of most recent report:\n ${dateStringFormatter(latestDate)}\n\n`
+    : "";
+  const datasetPublishedToolTip =
+    "Dataset published on Dataland on:\n " + convertUnixTimeInMsToDateString(singleDataAndMetaInfo.metaInfo.uploadTime);
```
Comment (jhaerDfine) [inline dataland-frontend/src/components/resources/dataTable/MultiLayerDataTable.vue:108]: mit dem doppelt "on" finde ich sonderbar. Was meinst du zu "Publication date of the dataset on Dataland:"?

File: dataland-internal-storage/src/main/kotlin/db/migration/V17__MigrateReferencedReports.kt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ * and also renames the reportDate field to publicationDate
+ */
+class V17__MigrateReferencedReports : BaseJavaMigration() {
+
+    private val frameworksToMigrate = listOf(
+        "eutaxonomy-non-financials",
+        "eutaxonomy-financials",
+        "sfdr",
+        "sme",
+    )
```
Comment (jhaerDfine) [inline dataland-internal-storage/src/main/kotlin/db/migration/V17__MigrateReferencedReports.kt:20]: double check, vsme?

File: dataland-internal-storage/src/main/kotlin/db/migration/V17__MigrateReferencedReports.kt [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            "sfdr" -> {
+                val general = dataset.getOrJavaNull("general") as JSONObject?
+                val generalGeneral = general?.getOrJavaNull("general") as JSONObject?
+                generalGeneral?.getOrJavaNull("referencedReports") as JSONObject?
+            }
+            "sme" -> {
+                val general = dataset.getOrJavaNull("general") as JSONObject?
+                val basicInformation = general?.getOrJavaNull("basicInformation") as JSONObject?
+                basicInformation?.getOrJavaNull("referencedReports") as JSONObject?
+            }
```
Comment (jhaerDfine) [inline dataland-internal-storage/src/main/kotlin/db/migration/V17__MigrateReferencedReports.kt:67]: sme wird migriert, hat also referenced reports? aber es gibt keine tooltips/referenced reports list für sme?

File: dataland-internal-storage/src/test/resources/db/migration/V17/originalNonFinancial.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+      "provider": null,
+      "dataSource": null
+    },
+    "referencedReports": {
+      "SustainabilityReport": {
+        "fileReference": "50a36c418baffd520bb92d84664f06f9732a21f4e2e5ecee6d9136f16e7e0b63",
+        "fileName": "SustainabilityReport",
+        "isGroupLevel": null,
+        "reportDate": null
+      }
```
Comment (jhaerDfine) [inline dataland-internal-storage/src/test/resources/db/migration/V17/originalNonFinancial.json:15]: sollte der nicht ein currency field haben? oder zumindestens einen Report ergänzen der alle felder hat?

File: dataland-internal-storage/src/test/resources/db/migration/V17/originalFinancial.json [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    "provider": "Daugherty - Lind",
+    "dataSource": {
+      "page": 132,
+      "fileName": "AnnualReport",
+      "fileReference": "50a36c418baffd520bb92d84664f06f9732a21f4e2e5ecee6d9136f16e7e0b63",
+      "tagName": "technologies"
+    }
+  },
+  "numberOfEmployees": 5530,
+  "referencedReports": null
```
Comment (jhaerDfine) [inline dataland-internal-storage/src/test/resources/db/migration/V17/originalFinancial.json:92]: sollte beim tests nicht zumindestens auch ein datensatz mit referenced reports vorhanden sein?

File: dataland-internal-storage/src/test/resources/db/migration/V17/originalNonFinancial.json (hunk 6 above)
Comment (aliabdoli77) [reply dataland-internal-storage/src/test/resources/db/migration/V17/originalNonFinancial.json:15]: this is for the (non-normal) case that such field didn't even exist in the first place. Its to make sure the migration doesn't have any weird behavior

File: dataland-internal-storage/src/test/resources/db/migration/V17/originalFinancial.json (hunk 7 above)
Comment (aliabdoli77) [reply dataland-internal-storage/src/test/resources/db/migration/V17/originalFinancial.json:92]: Technically there's no need to test the same scenarios for different frameworks. I used this test to make sure two things: what happens when the field is null - and if the path to referencedReports of this framework exists where im expecting it

**Pull request merged**